### PR TITLE
add search bar/modify filter for user

### LINKED ISSUE #7: Search bar for users
Add a search bar for users by email.

### PR DESCRIPTION
**Pull request by BaiasPaul**
add search bar/modify filter for user and make them work simultaneously

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -30,47 +34,81 @@ public function __construct(RendererInterface $renderer, UserService $questionIn
...
+        $email = $this->userService->getFromParameter('email', $request, "");
+        $role = $this->userService->getFromParameter('role', $request, "");
+        $currentPage = (int)$this->userService->getFromParameter('page', $request, 1);
+        $totalResults = (int)$this->userService->getEntityNumberOfPagesByField(User::class, ['email' => $email, 'role' => $role]);
+        $resultsPerPage = 5;
+
+        $paginator = new Paginator($totalResults, $currentPage, $resultsPerPage);
+        $paginator->setTotalPages($totalResults, $resultsPerPage);
+
+        $arguments['email'] = $email;
```
Comment (calinpristavu) [inline src/Controller/UserController.php]: you are building this array just to send it to the render method.you could always:
```
return $this->renderer->renderView("admin-users-listing.phtml", [
  'email' => $email
]);

File: src/Controller/UserController.php [hunk 2]
```diff
@@ -30,47 +34,81 @@ public function __construct(RendererInterface $renderer, UserService $questionIn
...
+     */
+    public function createUser(Request $request): Response
     {
         $name = $request->getParameter('name');
         $email = $request->getParameter('email');
         $password = $request->getParameter('password');
         $role = $request->getParameter('role');
         $this->userService->saveUser($name, $email, $password, $role);
-        $location = 'Location: http://quizApp.com/admin-users-listing?page=1';
+        $location = 'Location: http://quizApp.com/admin-users-listing';
```
Comment (calinpristavu) [inline src/Controller/UserController.php]: do not define a variable if you only use it once.

**Review by calinpristavu — CHANGES_REQUESTED** [review]

File: src/Controller/UserController.php (hunk 1 above)
Comment (BaiasPaul) [reply src/Controller/UserController.php]: $arguments array holds all the data(email, username, dropdownRole, paginator and users) not just email.
Are you suggesting something like this? 

`return $this->renderer->renderView("admin-users-listing.phtml", [
'email' => $email,
'usarname'=>$this->userService->getName(),
'dropdownRole' => $role,
'paginator' => $paginator,
'users' => $this->userService->getEntitiesByField(User::class, ['email' => $email, 'role' => $role], $currentPage, resultsPerPage)
]);`

Comment (calinpristavu) [reply src/Controller/UserController.php]: with proper formatting, yes
```
return $this->renderer->renderView("admin-users-listing.phtml", [ 
    'email' => $email, 
    'usarname'=>$this->userService->getName(), 
    'dropdownRole' => $role, 
    'paginator' => $paginator, 
    'users' => $users,
]);
```

**Pull request merged**